solarforecastarbiter/reports/figures/plotly_figures.py

```diff
@@ -296,10 +297,19 @@ def bar_subdivisions(df, category, metric):
             x_values = plot_data['index']
         else:
             x_values = []
+        if category == 'weekday':
+            # Fill with mon-fri values and pass to enforce displaying the full
+            # week of data.
+            y_values = [plot_data[plot_data['index'] == day]['value'].iloc[0]
+                        if not plot_data[plot_data['index'] == day].empty
+                        else 0 for day in x_ticks]
```

Comment:
can we use `nan` instead of 0?

Reply:
will the same problem occur with `hourly` if we exclude nighttime values?

Reply:
The hourly plots work correctly when dropping nighttime values. My test report excludes them. For some reason, plotly respects the x_range of an axis when provided as a series of numerical values, but not for things it decides are categorical.